src/exp_log_geo.jl

```diff
@@ -17,27 +16,44 @@ function exp(M::AbstractManifold, p, X)
     exp!(M, q, p, X)
     return q
 end
-function exp(M::AbstractManifold, p, X, t::Number)
+
+@doc raw"""
+    expt(M::AbstractManifold, p, X, t::Number = 1)
+
+Compute the exponential map of tangent vector `X` scaled by `t` at point `p`.
+Compared to [`exp`](@ref), this method provides the opportunity to
+avoid the allocation when computing `t*X`.
+By default, this method allocates the resulting point `q` and passes to [`expt!`](@ref).
+"""
+function expt(M::AbstractManifold, p, X, t::Number)
     q = allocate_result(M, exp, p, X, t)
-    exp!(M, q, p, X, t)
+    expt!(M, q, p, X, t)
     return q
 end
 
 """
     exp!(M::AbstractManifold, q, p, X)
-    exp!(M::AbstractManifold, q, p, X, t::Number = 1)
 
 Compute the exponential map of tangent vector `X`, optionally scaled by `t`,  at point `p`
 from the manifold [`AbstractManifold`](@ref) `M`.
 The result is saved to `q`.
 
-If you want to implement exponential map for your manifold, you should implement the
-method with `t`, that is `exp!(M::MyManifold, q, p, X, t::Number)`.
+If you want to implement exponential map for your manifold, you should implement the in-place
+method with, that is `expt!(M::MyManifold, q, p, X)`.
 
 See also [`exp`](@ref).
 """
 exp!(M::AbstractManifold, q, p, X)
-exp!(M::AbstractManifold, q, p, X, t::Number) = exp!(M, q, p, t * X)
+
+"""
+    expt!(M::AbstractManifold, q, p, X, t::Number = 1)
```

Comment:
```suggestion
    expt!(M::AbstractManifold, q, p, X, t::Number)
```
I think we don't need the default value for this function.